internal/databases/postgres/backup_push_handler.go

```diff
@@ -169,6 +169,9 @@ func (bh *BackupHandler) createAndPushBackup(ctx context.Context) {
 
 	err = bh.startBackup()
 	tracelog.ErrorLogger.FatalOnError(err)
+	err = bh.checkDataChecksums()
```

Comment:
I think we'd better do it before calling start backup

Reply:
> I think we'd better do it before calling start backup

I intentionally placed bh.checkDataChecksums() after bh.startBackup() because startBackup() initializes bh.Workers.QueryRunner and attempts to acquire the necessary locks (LOCK bh.Workers.QueryRunner.GetLockingPID). 

Moving checkDataChecksums() earlier would require additional initialization code for these components, which I wanted to avoid to keep the code streamlined. 

**Do you still prefer to move bh.checkDataChecksums() higher?**